app/javascript/src/Common/components/ToolbarSearch.jsx

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import * as React from 'react'
+
+import {
+  Button,
+  ButtonVariant,
+  Form,
+  InputGroup,
+  TextInput
+} from '@patternfly/react-core'
+import { SearchIcon } from '@patternfly/react-icons'
+
+import { createReactWrapper } from 'utilities'
+
+import './ToolbarSearch.scss'
+
+type Props = {
+  placeholder: string
+}
+
+const ToolbarSearch = ({ placeholder }: Props): React.Node => {
+  const formId = 'toolbar-search-form'
+  const query = new URL(window.location).searchParams.get('search[query]')
+  const [searchText, setSearchText] = React.useState<string>(query || '')
+
+  const ref = React.useRef()
+
+  React.useEffect(() => {
+    const input = ref.current
+    if (input) input.addEventListener('search', handleOnSearch)
+
+    // $FlowIgnore[incompatible-use] should not be null at this point
+    return () => input.removeEventListener('search', handleOnSearch)
+  }, [])
+
+  const onSubmitSearch = (value: string) => {
+    const form: HTMLFormElement = document.forms[formId]
+
+    if (!query && value.length === 0) {
+      return
+    } else if (value.length === 0) {
+      removeEmptySearchQueryFromURL(form)
+    } else if (value.length < 3) {
+      // Sphinx does not index less than 3 characters. Prevent form from being submitted.
+      return
+    }
+
+    form.submit()
+  }
+
+  const removeEmptySearchQueryFromURL = (form: HTMLFormElement) => {
+    form.elements['search[query]'].removeAttribute('name')
+    form.elements['utf8'].removeAttribute('name')
+  }
+
+  const handleOnSearch = (e: SyntheticEvent<HTMLInputElement>) => {
+    e.preventDefault()
+    onSubmitSearch(e.currentTarget.value)
+  }
+
+  return (
+    <Form
+      id={formId}
+      acceptCharset="UTF-8"
+      method="get"
+      role="search"
+      onSubmit={e => e.preventDefault()}
+    >
+      <InputGroup>
+        <input name="utf8" type="hidden" value="✓" />
+        <TextInput
+          // $FlowIgnore[incompatible-type] it's fine, really
+          ref={ref}
+          placeholder={placeholder}
+          name="search[query]"
+          type="search"
+          aria-label="Search"
+          value={searchText}
+          onChange={setSearchText}
+          autoComplete="off"
+        />
+        <Button
+          variant={ButtonVariant.control}
+          aria-label="search button for search input"
+          isDisabled={searchText.length > 0 && searchText.length < 3}
```

Comment:
```suggestion
          isDisabled={searchText.length < 3}
```
If you disable only when `searchText.length > 0` the button is clickable with empty input. 

And it's a bit strange that the button is clickable when first rendered, but becomes disabled after the first character, then is clickable again with 3 chars.
I think it's better if first rendered as disabled, and only enabled when reaching 3 chars.

Reply:
However, we need to allow users to submit an "empty" search in order to show all results. If we disable the button with 0 chars, then they will be forced to refresh the page.

Reply:
We could disable it as long as the search is invalid, for instance when searching the same thing twice. Thoughts @cntlsn ?

Reply:
> However, we need to allow users to submit an "empty" search in order to show all results. If we disable the button with 0 chars, then they will be forced to refresh the page.

How is that? Show all results (index page) is a user triggered action?

Reply:
Resolving as we've discussed it elsewhere: button will be always enabled and it will show a popover with the message "needs 3 chars to search".